internal/controllers/vehicle_subscription_controller.go

```diff
@@ -231,8 +294,102 @@ func (v *VehicleSubscriptionController) SubscribeAllVehiclesToWebhook(c *fiber.C
+// @Success      201 {object} map[string]string "Successfully subscribed X vehicles to the webhook"
+// @Failure      400 {object} map[string]string "Event ID and vehicle token IDs are required"
+// @Failure      401 {object} map[string]string "Unauthorized"
+// @Failure      500 {object} map[string]string "Internal server error"
+// @Router       /subscriptions/{eventID} [post]
+func (v *VehicleSubscriptionController) SubscribeMultipleVehiclesToWebhook(c *fiber.Ctx) error {
+	eventID := c.Params("eventID")
+	if eventID == "" {
+		return c.Status(http.StatusBadRequest).JSON(fiber.Map{"error": "Event ID is required"})
+	}
+	vehicleIDsParam := c.Query("vehicleTokenIDs")
+	if vehicleIDsParam == "" {
+		return c.Status(http.StatusBadRequest).JSON(fiber.Map{"error": "Vehicle token IDs are required"})
+	}
+	ids := strings.Split(vehicleIDsParam, ",")
+	devLicense, ok := c.Locals("developer_license_address").([]byte)
+	if !ok {
+		v.logger.Error().Msg("Developer license not found in request context")
+		return c.Status(http.StatusUnauthorized).JSON(fiber.Map{"error": "Unauthorized"})
+	}
+	var successCount int
+	for _, idStr := range ids {
+		idStr = strings.TrimSpace(idStr)
+		tokenDecimal := types.Decimal{}
+		if err := tokenDecimal.Scan(idStr); err != nil {
+			v.logger.Error().Err(err).Msgf("Invalid token ID format for vehicle %v", idStr)
+			continue
+		}
+		eventVehicle := &models.EventVehicle{
+			VehicleTokenID:          tokenDecimal,
+			EventID:                 eventID,
+			DeveloperLicenseAddress: devLicense,
+			CreatedAt:               time.Now(),
+			UpdatedAt:               time.Now(),
+		}
+		if err := eventVehicle.Insert(c.Context(), v.store.DBS().Writer, boil.Infer()); err != nil {
+			v.logger.Error().Err(err).Msgf("Failed to subscribe vehicle %v to webhook", idStr)
+			continue
+		}
+		successCount++
+	}
 	return c.Status(http.StatusCreated).JSON(fiber.Map{
 		"message": fmt.Sprintf("Successfully subscribed %d vehicles to the webhook", successCount),
 	})
 }
+
+// UnshareVehiclePermissions godoc
+// @Summary      Remove all subscriptions for a vehicle when permissions are revoked
+// @Description  Removes all webhook subscriptions for a given vehicle token when the vehicle's permissions are un-shared by the user.
+// @Tags         Vehicle Subscriptions
+// @Produce      json
+// @Param        vehicleTokenID path string true "Vehicle Token ID"
+// @Param        Authorization header string true "Bearer token"
+// @Success      200 {object} map[string]string "Webhook subscriptions removed successfully"
+// @Failure      400 {object} map[string]string "Invalid vehicle token ID"
+// @Failure      401 {object} map[string]string "Unauthorized"
+// @Failure      500 {object} map[string]string "Internal server error"
+// @Router       /subscriptions/{vehicleTokenID}/unshare [delete]
+func (v *VehicleSubscriptionController) UnshareVehiclePermissions(c *fiber.Ctx) error {
+	vehicleTokenIDStr := c.Params("vehicleTokenID")
```

Comment:
Similarly, checks performed on most controller functions can be abstracted if they repeat more than once.